spec/serializers/current_user_serializer_spec.rb

```diff
@@ -257,11 +257,48 @@
     end
   end
 
+  describe '#sidebar_tags' do
+    fab!(:tag_1) { Fabricate(:tag, name: "foo") }
+    fab!(:tag_2) { Fabricate(:tag, name: "bar") }
+    fab!(:hidden_tag) { Fabricate(:tag, name: "secret") }
+    fab!(:staff_tag_group) { Fabricate(:tag_group, permissions: { "staff" => 1 }, tag_names: ["secret"]) }
+
+    it 'includes visible default sidebar tags' do
+      SiteSetting.enable_experimental_sidebar_hamburger = true
+      SiteSetting.tagging_enabled = true
+      SiteSetting.default_sidebar_tags = "foo|bar|secret"
+
+      json = serializer.as_json
+
+      expect(json[:sidebar_tags]).to eq([
+        { name: "foo", pm_only: false },
+        { name: "bar", pm_only: false }
+      ])
+    end
+
+    it 'includes tags choosen by user' do
+      SiteSetting.enable_experimental_sidebar_hamburger = true
+      SiteSetting.tagging_enabled = true
+      SiteSetting.default_sidebar_tags = "foo|bar|secret"
+      tag_sidebar_section_link = Fabricate(:tag_sidebar_section_link, user: user)
+
+      json = serializer.as_json
+
+      expect(json[:sidebar_tags]).to eq([
+        { name: tag_sidebar_section_link.linkable.name, pm_only: false }
+      ])
+    end
+  end
+
   describe '#sidebar_category_ids' do
-    fab!(:category_sidebar_section_link) { Fabricate(:category_sidebar_section_link, user: user) }
-    fab!(:category_sidebar_section_link_2) { Fabricate(:category_sidebar_section_link, user: user) }
+    fab!(:category) { Fabricate(:category) }
+    fab!(:category_2) { Fabricate(:category) }
+    fab!(:private_category) { Fabricate(:private_category, group: Fabricate(:group)) }
+    let(:category_sidebar_section_link) { Fabricate(:category_sidebar_section_link, user: user) }
+    let(:category_sidebar_section_link_2) { Fabricate(:category_sidebar_section_link, user: user) }
```

Comment:
Is there a reason we're using `let` instead of `fab!` here?

Reply:
yes, I don't want those instances to be created upfront to test that default SiteSetting categories are working. And if link records are created, they take precedence over default SiteSetting

Reply:
got it :+1: